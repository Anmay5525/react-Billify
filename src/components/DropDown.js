import React from "react";
import PropTypes from "prop-types";

export default function DropDown({ list, handleSelect }) {
  return (
    <select
      style={{
        width: "100%",
        height: "26px",
      }}
      onChange={handleSelect}
    >
      <option className="option">Select Item</option>
      {list.map((item) => {
        return (
          <option
            className="option"
            key={item.name + window.performance.now()}
            value={JSON.stringify(item)}
          >
            {item.name}
          </option>
        );
      })}
    </select>
  );
}

DropDown.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleSelect: PropTypes.func.isRequired,
};
