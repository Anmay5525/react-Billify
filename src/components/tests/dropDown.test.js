import React from 'react';
import { configure, shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DropDown from '../DropDown';

configure({adapter: new Adapter()});

describe('<Dropdown/>',()=>{
    it('renders correctly (for two items in list)',()=>{
        const tree=renderer.create(
        <DropDown
            list={[{name:"name1"},{name:"name2"}]}
            handleSelect={()=>{}}
        />)
        expect(tree).toMatchSnapshot();
    });
    it('renders correctly for empty array as list',()=>{
        const tree=renderer.create(
        <DropDown
            list={[]}
            handleSelect={()=>{}}
        />)
        expect(tree).toMatchSnapshot();
    });
})