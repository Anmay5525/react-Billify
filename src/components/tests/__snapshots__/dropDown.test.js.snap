// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown/> renders correctly (for two items in list) 1`] = `
<select
  onChange={[Function]}
  style={
    Object {
      "height": "26px",
      "width": "100%",
    }
  }
>
  <option
    className="option"
  >
    Select Item
  </option>
  <option
    className="option"
    value="{\\"name\\":\\"name1\\"}"
  >
    name1
  </option>
  <option
    className="option"
    value="{\\"name\\":\\"name2\\"}"
  >
    name2
  </option>
</select>
`;

exports[`<Dropdown/> renders correctly for empty array as list 1`] = `
<select
  onChange={[Function]}
  style={
    Object {
      "height": "26px",
      "width": "100%",
    }
  }
>
  <option
    className="option"
  >
    Select Item
  </option>
</select>
`;
